tests/characters/test_alice_can_grant_and_revoke.py

```diff
@@ -1,36 +1,56 @@
 import datetime
 
+from apistar.test import TestClient
+
 from nkms.characters import Ursula
 from nkms.crypto.api import keccak_digest
+from nkms.crypto.constants import PUBLIC_KEY_LENGTH
 from nkms.crypto.powers import SigningPower, EncryptingPower
 from nkms.crypto.utils import BytestringSplitter
-from nkms.keystore.keypairs import PublicKey
 from tests.utilities import MockNetworkyStuff
-from apistar.test import TestClient
+from umbral.fragments import KFrag
+from umbral.keys import UmbralPublicKey
 
 
 def test_grant(alice, bob, ursulas):
     networky_stuff = MockNetworkyStuff(ursulas)
     policy_end_datetime = datetime.datetime.now() + datetime.timedelta(days=5)
     n = 5
     uri = b"this_is_the_path_to_which_access_is_being_granted"
-    policy = alice.grant(bob, uri, networky_stuff, m=3, n=n, expiration=policy_end_datetime)
+    policy = alice.grant(bob, uri, networky_stuff, m=3, n=n,
+                         expiration=policy_end_datetime)
 
     # The number of policies is equal to the number of Ursulas we're using (n)
     assert len(policy._accepted_contracts) == n
 
     # Let's look at the first Ursula.
     ursula = list(policy._accepted_contracts.values())[0].ursula
```

Comment:
A call to `next()` may suffice here instead of casting to a list.

Reply:
👍

Reply:
Actually, we (sadly and frustratingly) can't do that here, because this is a `dict_values` and not an iterator.

Reply:
✔️